pallets/domains/src/lib.rs

```diff
@@ -0,0 +1,517 @@
+
+    use frame_system::pallet_prelude::*;
+
+    use sp_runtime::traits::{Saturating, StaticLookup, Zero};
+    use sp_std::{cmp::Ordering, convert::TryInto, vec::Vec};
+
+    use pallet_parachain_utils::ensure_content_is_valid;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config + pallet_timestamp::Config {
+        /// The overarching event type.
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        /// The currency trait.
+        type Currency: ReservableCurrency<<Self as frame_system::Config>::AccountId>;
+
+        /// Domains minimum length.
+        #[pallet::constant]
+        type MinDomainLength: Get<u32>;
+
+        /// Domains maximum length.
+        #[pallet::constant]
+        type MaxDomainLength: Get<u32>;
+
+        /// Maximum amount of domains that can be registered per account.
+        #[pallet::constant]
+        type MaxDomainsPerAccount: Get<u32>;
+
+        /// The maximum domains amount can be inserted to a storage at once.
+        #[pallet::constant]
+        type DomainsInsertLimit: Get<u32>;
+
+        /// The maximum amount of time the domain may be held for.
+        #[pallet::constant]
+        type RegistrationPeriodLimit: Get<<Self as frame_system::pallet::Config>::BlockNumber>;
+
+        /// The length limit for the domains meta outer value.
+        #[pallet::constant]
+        type MaxOuterValueLength: Get<u32>;
+
+        /// The amount held on deposit for storing the domains structure.
+        #[pallet::constant]
+        type BaseDomainDeposit: Get<BalanceOf<Self>>;
+
+        /// The amount held on deposit per byte within the domains outer value.
+        #[pallet::constant]
+        type OuterValueByteDeposit: Get<BalanceOf<Self>>;
+
+        /// Weight information for extrinsics in this pallet.
+        type WeightInfo: WeightInfo;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub (super) trait Store)]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(_);
+
+    #[pallet::storage]
+    #[pallet::getter(fn is_domain_reserved)]
+    pub(super) type ReservedDomains<T: Config> =
+        StorageMap<_, Blake2_128Concat, DomainName<T>, bool, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn registered_domain)]
+    pub(super) type RegisteredDomains<T: Config> =
+        StorageMap<_, Blake2_128Concat, DomainName<T>, DomainMeta<T>>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn domains_by_owner)]
+    pub(super) type DomainsByOwner<T: Config> =
+        StorageMap<_,
+            Twox64Concat,
+            T::AccountId,
+            BoundedVec<DomainName<T>, T::MaxDomainsPerAccount>,
+            ValueQuery,
+        >;
+
+    #[pallet::storage]
+    pub(super) type DomainByInnerValue<T: Config> =
+        StorageMap<_, Blake2_128Concat, DomainInnerLink<T::AccountId>, DomainName<T>>;
+
+    #[pallet::storage]
+    pub(super) type AllowedTlds<T: Config> =
```

Comment:
```suggestion
    pub(super) type SupportedTlds<T: Config> =
```

Example of copywriting from Namecheap:
<img width="319" alt="image" src="https://user-images.githubusercontent.com/153928/160589919-ac05bc70-d688-425d-9888-d3684e8cdff8.png">